testing/test_gateway.py

```diff
@@ -645,3 +645,35 @@ def test_main_thread_only_concurrent_remote_exec_deadlock(
             ch.close()
         gw.exit()
         gw.join()
+
+
+def test_concurrent_rinfo(
+    execmodel: gateway_base.ExecModel, makegateway: Callable[[str], Gateway]
+) -> None:
+    gw = makegateway(f"execmodel={execmodel.backend}//popen")
```

Comment:
It seems this test fails because the Group makegateway method uses the execmodel from the group which is "thread"  even though spec.exedmodel is "main_thread_only":
```python
        elif spec.popen or spec.ssh or spec.vagrant_ssh:
            io = gateway_io.create_io(spec, execmodel=self.execmodel)
            gw = gateway_bootstrap.bootstrap(io, spec)
```

Reply:
I'm not sure what the base course is to fix this test, but anyway it passes if I force main_thread_only when necessary like this:
```diff
--- a/src/execnet/multi.py
+++ b/src/execnet/multi.py
@@ -152,3 +152,10 @@ class Group:
         elif spec.popen or spec.ssh or spec.vagrant_ssh:
-            io = gateway_io.create_io(spec, execmodel=self.execmodel)
+            io = gateway_io.create_io(
+                spec,
+                execmodel=(
+                    self.execmodel
+                    if getattr(spec, "execmodel", None) != "main_thread_only"
+                    else get_execmodel(spec.execmodel)
+                ),
+            )
             gw = gateway_bootstrap.bootstrap(io, spec)
```

Reply:
I've changed the test to instantiate an execnet.Group with the desired execmodel, and that works nicely.

Reply:
> Looks like something that would benefit from having an intentional test?

Now `test_concurrent_rinfo` addresses the missing test coverage for concurrent _rinfo calls.

> And maybe some smoke tests with `pytest-cov` and `pytest-xdist` in the same env?

I suppose we could automate smoke tests against reverse dependencies like this, but anyway I ran a manual smoke test with one of my projects and pytest-xdist + pytest-cov and it successfully produced coverage:
```
Name                          Stmts   Miss  Cover   Missing
-----------------------------------------------------------
src/pyrqlite/__init__.py          0      0   100%
src/pyrqlite/_ephemeral.py       81      9    89%   14-16, 61-62, 81-85, 94
src/pyrqlite/connections.py     107     30    72%   10-12, 16-18, 65, 70-75, 89-94, 100-104, 130-143, 187-190, 193-198
src/pyrqlite/constants.py         8      0   100%
src/pyrqlite/cursors.py         210     22    90%   13-15, 28-29, 50, 53, 74, 84, 116, 184-185, 265, 267, 274-275, 280, 283-284, 309, 312, 315, 318
src/pyrqlite/dbapi2.py           20      0   100%
src/pyrqlite/exceptions.py        1      0   100%
src/pyrqlite/extensions.py      121     13    89%   33, 48-60, 141, 143, 190
src/pyrqlite/row.py              38      5    87%   5-6, 45, 76, 79
src/pyrqlite/types.py            11      0   100%
-----------------------------------------------------------
```

Reply:
I was able to reproduce the issue using tox with setuptools, and also this PR applied to the .tox/py virtualenv tox was successful and pytest-cov produced results.